core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -633,14 +637,13 @@ private void logStopFailure(Throwable cause) {
 
         @Override
         public void operationComplete(ChannelFuture f) {
-            final Channel ch = f.channel();
+            final ServerChannel ch = (ServerChannel) f.channel();
             assert ch.eventLoop().inEventLoop();
 
-            if (f.isSuccess()) {
```

Comment:
Is the logic change this one? The others appear to be no-op to me. We used to close using a set of channels and calling `ch.eventLoop().parent ()` but now store that as keys of a map. If I'm not mistaken in it being the same as before, I might consider the previous version with the `Set` to be easier to reason about.

Reply:
Good point. Made `serverChannels` a `Set<ServerChannel>`.